core/services/offchainreporting/oracle.go

```diff
@@ -210,19 +214,35 @@ type dataSource struct {
 
 var _ ocrtypes.DataSource = (*dataSource)(nil)
 
+// The context passed in here has a timeout of observationTimeout.
+// Gorm/pgx doesn't return a helpful error upon cancellation, so we manually check for cancellation and return a
+// appropriate error.
 func (ds dataSource) Observe(ctx context.Context) (ocrtypes.Observation, error) {
+	start := time.Now()
 	runID, err := ds.pipelineRunner.CreateRun(ctx, ds.jobID, nil)
+	endCreate := time.Now()
+	if ctx.Err() != nil {
+		return nil, errors.Errorf("context cancelled due to timeout or shutdown, cancel create run. Runtime %v", endCreate.Sub(start))
+	}
 	if err != nil {
+		logger.Errorw("Error creating new pipeline run", "jobID", ds.jobID, "error", err)
```

Comment:
Does it not get logged elsewhere?

Reply:
No I pulled it up here to only log it in the case that the error is not caused by using a stale txContext